### ISSUE TITLE
Pivot table cannot validly group rows and keep ordering

### ISSUE THREAD

**Issue by comdiv** [apache/superset#20564]
While we have only one `row ordering` field but can add multiple row dimensions it's assumed that when we 
set `order by value decending` it will be work same way as with `order by a-z`:
1. first row as group is ordered by a-z
2. then second row inside group ordering by a-z

But if you choose order by value - it's breaking all groupping - all group rows occures multiple times in pivot!

Hard to explain how to exactly reproduce it.

Just compre two variances of ordering `a-z` and `by value`

![image](https://user-images.githubusercontent.com/2387862/176914921-b25cc77e-df5a-4ee4-90d4-dd0943cb0843.png)

As you can see - it cause valid behavior - group ordering then subgroup ordering

But try to set another option

![image](https://user-images.githubusercontent.com/2387862/176914756-d6d8cf5c-d5eb-4b72-9ef1-08d643d5b288.png)

It breaks table! Group is splitting and Subtotals not counted as expected!!!!

**Comment by comdiv** [comment]
superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/utilities.js:694
```
  case 'value_a_to_z':
          this.rowKeys.sort((a, b) => naturalSort(v(a, []), v(b, [])));
          break;
        case 'value_z_to_a':
          this.rowKeys.sort((a, b) => -naturalSort(v(a, []), v(b, [])));
          break;
```
doesn't reqalculate subtotals so cannot sort them correctly

**Comment by rusackas** [comment]
Just to drop a note here, I'm trying to get an ephemeral environment up to test the PR. Hopefully we can get this merged, but more pivot table expertise would be nice for testing.

**Comment by rusackas** [comment]
Assuming this is still an issue in current Superset? I'm trying _again_ to spin up a test environment, but hoping @kgabryje can help test/review as well.

**Comment by JZ6** [comment]
yes it is still an issue, but we implemented a fix for it, maybe will submit a pr soon

**Comment by rusackas** [comment]
Probably still an issue, but we're open to updated context and/or a PR if anyone is interested.

**Comment by rusackas** [comment]
This hasn't _really_ been touched in quite some time. We're still open to volunteers, but this may be closed in time as "not planned" to inactivity.

**Comment by rusackas** [comment]
Closing this one in favor of https://github.com/apache/superset/issues/23353
